client/src/app/pages/dashboard/maps/map-info/map-leaderboard/map-leaderboard.component.ts

```diff
@@ -17,30 +17,53 @@ export class MapLeaderboardComponent implements OnInit {
   @Input('mapID')
   set mapID(value: number) {
     this._mapID = value;
-    this.loadLeaderboardRuns(this._mapID);
+    this.loadLeaderboardRuns();
   }
   filterActive: boolean;
   leaderboardRanks: UserMapRank[];
   searchedRanks: boolean;
+  filterOption: number;
```

Comment:
This should be renamed to something better:

```suggestion
  filterLeaderboardType: LeaderboardType;
```